### PR TITLE
(CDAP-3724) Move lineage serde classes to cdap-proto, so they can be …

### PR DESCRIPTION
**Pull request by bdmogal**
…accessed by REST API clients.

Build: http://builds.cask.co/browse/CDAP-RBT489

### CODE REVIEW

**Comment by poornachandra** [comment]
Made sure that the JSON representation of LineageRecord has not changed after this change.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageSerializer.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  private LineageSerializer() {}
+
+  public static LineageRecord toLineageRecord(long start, long end, Lineage lineage) {
+    Set<RelationRecord> relationBuilder = new HashSet<>();
+    Map<String, ProgramRecord> programBuilder = new HashMap<>();
+    Map<String, DataRecord> dataBuilder = new HashMap<>();
+
+    for (Relation relation : lineage.getRelations()) {
+      System.out.println("!!!!!!!! GOT RELATION = " + relation);
```
Comment (bdmogal) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageSerializer.java]: remove

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-RBT489-3

**Comment by bdmogal** [comment]
LGTM, as long as the REST API output has been tested to not have changed.

**Comment by poornachandra** [comment]
Verified JSON output has not changed for lineage.

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**